### ISSUE TITLE
vmselect does not allow federate metrics when one fo storage node is down

### ISSUE THREAD

**Issue by gasgurda** [VictoriaMetrics/VictoriaMetrics#5096]
### Describe the bug

When one of the nodes unavailable, scraping metrics doesn't work vi federate:
"...cannot export federated metrics, because some of vmstorage nodes are unavailable"
I have vmstorage cluster with 3 storage nodes
replicationFactor - 2

vmstorage should not stop serving federate requests when one storage node unavailable

### To Reproduce

Stop one storage node. Make a federate request

### Version

v1.93.1
v1.93.5

**Comment by zekker6** [comment]
Hello @gasgurda 

Could you please share flags set for vmselect nodes?

I've tested this setup and in case `replicationFactor` flag is used at vmselect and set to 2 than vmselect should be able to process federate request.

**Comment by gasgurda** [comment]
- -cacheDataPath=/select-cache
        - -dedup.minScrapeInterval=25s
        - -httpListenAddr=:8481
        - -search.logSlowQueryDuration=10s
        - -search.maxConcurrentRequests=20
        - -selectNode=vmselect-victoria-metrics-k8s-stack-0.vmselect-victoria-metrics-k8s-stack.monitoring:8481,vmselect-victoria-metrics-k8s-stack-1.vmselect-victoria-metrics-k8s-stack.monitoring:8481
        - -storageNode=vmstorage-victoria-metrics-k8s-stack-0.vmstorage-victoria-metrics-k8s-stack.monitoring:8401,vmstorage-victoria-metrics-k8s-stack-1.vmstorage-victoria-metrics-k8s-stack.monitoring:8401,vmstorage-victoria-metrics-k8s-stack-2.vmstorage-victoria-metrics-k8s-stack.monitoring:8401
        - -vmalert.proxyURL=http://vmalert-victoria-metrics-k8s-stack.monitoring.svc:8080


replicationFactor parameter was not set automaticaly via helm-chart victoria-metrics-k8s-stack (set just only in vminsert). I set replicationFactor  manually for vmselect and now it works. Thank you.

**Comment by zekker6** [comment]
@f41gh7 @Amper @Haleygo Is it intended that `replicationFactor` is not set by operator for `vmselect` nodes?
I'd suggest to set RF at `vmselect` as well since it now does not lead to unexpected results with the latest releases.

**Comment by gasgurda** [comment]
No, operator does not set RF for vmselect, it set RF only for vminsert. Tested on 1.35.1 and 1.38.0 versions. It will be nice if operator add this parameter into argument for vmselect.

**Comment by Amper** [comment]
I don't know how safe it is to change the default behavior of the vmcluster for all users?

@f41gh7 @hagen1778 @valyala wdyt?

**Comment by hagen1778** [comment]
@Amper I think it is safe applying replicationFactor to vmselects for versions >= https://docs.victoriametrics.com/CHANGELOG.html#v1913

See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1207
https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4538
